src/main/java/seedu/javaninja/Quiz.java

```diff
@@ -0,0 +1,84 @@
+package seedu.javaninja;
+
+import java.util.List;
+import java.util.Scanner;
+
+public class Quiz {
+    private Topic topic;
+    private int currentQuestionIndex;
+    private int correctAnswers;
+    private Scanner scanner;
+
+    public Quiz(Topic topic, Scanner scanner) {
+        assert topic != null : "Topic must not be null";
+        assert !topic.getQuestions().isEmpty() : "Topic must contain at least one question";
+        this.topic = topic;
+        this.scanner = scanner;
+        this.currentQuestionIndex = 0;
+        this.correctAnswers = 0;
+    }
+
+    public void start() {
+        List<Question> questions = topic.getQuestions();
+
+        if (questions.isEmpty()) {
+            throw new IllegalStateException("Cannot start a quiz with no questions.");
+        }
+
+        while (currentQuestionIndex < questions.size()) {
```

Comment:
looks quite nested. can you extract some code to individual methods and invoke them to reduce nesting?